python-shell/src/test/test_connector.py

```diff
@@ -21,247 +21,150 @@
+            elements = gc.execute_operation(
+                g.GetElements(
+                    input=[
+                        g.EntitySeed('M5:10')
+                    ],
+                    view=g.View(
+                        edges=[
+                            g.ElementDefinition(
+                                group='JunctionLocatedAt'
+                            )
+                        ]
+                    )
                 )
             )
-        )
-
-        self.assertEqual(
-            [g.Edge("JunctionLocatedAt", "M5:10", "390466,225615", True, {},
-                    "SOURCE")],
-            elements)
-
-    def test_class_initilisation(self):
-        """Test that the gaffer_connector class is correctly initialised with instance attributes"""
-        host = 'http://localhost:8080/rest/latest'
-        verbose = False
-        headers = {"dummy_Header": "value"}
-        gc = gaffer_connector.GafferConnector(host, verbose, headers)
-
-        actuals = [gc._host, gc._verbose, gc._headers]
-        expecteds = [host, verbose, headers]
-
-        for actual, expected in zip(actuals, expecteds):
-            self.assertEqual(actual, expected)
-
-    def test_raise_connection_error(self):
-        """Test that a ConnectionError is correctly raised when a HTTP 404 error is caught"""
-        # Define a host that has an invalid endpoint in order to get a HTTP 404 error
-        host_with_bad_endpoint = "http://localhost:8080/badEndPoint"
-        gc = gaffer_connector.GafferConnector(host_with_bad_endpoint)
 
-        # Check that a ConnectionError is raised (which is catching the underlying HTTP 404)
-        with self.assertRaises(ConnectionError):
-            gc.execute_get(g.GetOperations())
+            self.assertEqual(
+                [g.Edge("JunctionLocatedAt", "M5:10", "390466,225615", True, {},
+                        "SOURCE")],
+                elements)
+
+        def test_class_initilisation(self):
+            '''Test that the gaffer_connector class is correctly initialised with instance attributes'''
+            host = 'http://localhost:8080/rest/latest'
+            verbose = False
+            headers = {"dummy_Header": "value"}
+            gc = gaffer_connector.GafferConnector(host, verbose, headers, client_class=self.client_class)
+
+            actuals = [gc._host, gc._verbose, gc._headers]
+            expecteds = [host, verbose, headers]
+
+            for actual, expected in zip(actuals, expecteds):
+                self.assertEqual(actual, expected)
+
+        def test_raise_connection_error(self):
+            '''Test that a ConnectionError is correctly raised when a HTTP 404 error is caught'''
+            # Define a host that has an invalid endpoint in order to get a HTTP 404 error
+            host_with_bad_endpoint = "http://localhost:8080/badEndPoint"
```

Comment:
What happens when the connection is tried using HTTPS?

Reply:
Good question, there needs to be a lot more tests with Gaffer and gaffer-tools using https and some docs.
I didn't add this test in particular though, it was moved and the diff is weird.

Reply:
Perhaps a test to capture the current behaviour?

Reply:
I think that would be a good idea eventually but not really in scope of this PR, when I run the example rest locally a https endpoint isn't provided so I can't really test gafferpy with https. Maybe a Gaffer issue should be made to look into this.

Reply:
Perhaps a test to capture the current behaviour when the endpoint is incorrectly specified as an HTTPS endpoint?